deals/deals.go

```diff
@@ -92,7 +97,34 @@ func (m *Module) Store(ctx context.Context, waddr string, data io.Reader, dcfgs
 		}
 		proposals = append(proposals, *proposal)
 	}
-	return proposals, failed, nil
+	return dataCid, proposals, failed, nil
+}
+
+func (m *Module) Retrieve(ctx context.Context, waddr string, cid cid.Cid) (io.ReadCloser, error) {
```

Comment:
Quite simple `Retrieve` API: using a specified wallet, it tries to retrieve the data with `cid`.
The API fetches the data from any miner that has it. Unfortunately, not much complex things can be done *now* since Lotus only holds the last miner for which *we* stored the data. 

The bold part is important. No feature yet to discover if other Lotus node stored that cid... most probably will be implemented by Lotus in the near future. When that happens, I think this method would be refactored since we would like to have more fine-grained control in which _Retrieval Ask_ we want to use to fetch the data.